File: src/main/java/org/parham/seasonjob/commands/default_cmd/Kick.java
package org.parham.seasonjob.commands.default_cmd;

import org.bukkit.Bukkit;
import org.bukkit.OfflinePlayer;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.parham.seasonjob.commands.SeasonCommand;
import org.parham.seasonjob.data.job.Job;
import org.parham.seasonjob.data.job.JobManager;
import org.parham.seasonjob.data.member.Member;
import org.parham.seasonjob.data.member.MemberManager;
import org.parham.seasonjob.data.sender.Messages;

import java.util.List;

public class Kick implements SeasonCommand {
    @Override
    public boolean hasPermission() {
        return true;
    }

    @Override
    public boolean isPlayerOnly() {
        return false;
    }

    @Override
    public String getPermission() {
        return "seasonjob.kick";
    }

    @Override
    public int getNeedArguments() {
        return 1;
    }

    @Override
    public String getUsage() {
        return Messages.getMessage("usage-kick");
    }

    @Override
    public void execute(CommandSender sender, Command cmd, String label, String[] args, boolean isLeader, boolean hasAccess) {
        OfflinePlayer target = Bukkit.getOfflinePlayer(args[0]);


        if (!MemberManager.contains(target.getUniqueId())) {
            sender.sendMessage(Messages.getMessage("unemployed-kick"));
        } else if ((isLeader && !hasAccess) && (hasPermission() && !sender.hasPermission(getPermission() + "." + MemberManager.getMember(target.getUniqueId()).getJob()))) {
            sender.sendMessage(Messages.getMessage("deny-permission-kick"));
        } else {
            if (sender instanceof Player && ((Player) sender).getUniqueId() == target.getUniqueId()) {
                sender.sendMessage(Messages.getMessage("self-kick"));
            } else {
                Member member = MemberManager.getMember(target.getUniqueId());
                Job job = JobManager.getJob(member.getJob());
                if (!job.getLeader().getUUID().equals(target.getUniqueId()) || sender.hasPermission(getPermission() + ".leader")) {
                    MemberManager.delete(target.getUniqueId());
                    sender.sendMessage(Messages.getMessage("success-kick"));
                } else {
                    sender.sendMessage(Messages.getMessage("deny-permission-kick"));
                }
            }
        }
    }

    @Override
    public List<String> getCompletions(CommandSender sender, String[] args) {
        return null;
    }
}

File: src/main/kotlin/ir/parham/seasonJob/Commands/Default/Kick.kt
package ir.parham.seasonJob.Commands.Default

import Libs.API.ir.parham.SeasonJobsAPI.Commands
import Libs.API.ir.parham.SeasonJobsAPI.Senders.Message
import ir.parham.SeasonJobsAPI.Actions.Member
import org.bukkit.Bukkit
import org.bukkit.OfflinePlayer
import org.bukkit.command.CommandSender
import java.util.ArrayList
import kotlin.collections.List

class Kick : Commands {
    override fun runner(sender: CommandSender, args: Array<out String>) {
        val message = Message()
        if (args.size < 2) {
            // usage
            sender.sendMessage(null, "kickUsage")
        } else if (Bukkit.getOfflinePlayer(args[1]) == null) {
            // player not found
            sender.sendMessage(message.get(Bukkit.getOfflinePlayer(args[1]), "playerNotExits"))
        } else if (!Member().contains(Bukkit.getOfflinePlayer(args[1]).uniqueId)) {
            // is employed
            val player = Bukkit.getOfflinePlayer(args[1])
            sender.sendMessage(message.get(player, "playerIsEmploy"))
        } else if (!sender.hasPermission("seasonjobs.kick." + Member().get(Bukkit.getOfflinePlayer(args[1]).getUniqueId()))) {
            // deny perm
            val player = Bukkit.getOfflinePlayer(args[1])
            sender.sendMessage(message.get(player, "denyPerm"))
        } else {
            val player = Bukkit.getOfflinePlayer(args[1])
            remover(player)
            sender.sendMessage(message.get(player, "kickSuccess"))
            // kicked
        }
    }

    override fun completer(sender: CommandSender, args: Array<out String>): List<String> {
        if (args.size == 2) {
            val players: ArrayList<String> = ArrayList()
            for (player in Bukkit.getOnlinePlayers()) {
                players.add(player.name)
            }
            return players
        }
        return listOf("")
    }

    override fun remover(target: OfflinePlayer) {
        Member().remove(target.uniqueId)
    }

    override fun adder(target: OfflinePlayer) {
        TODO("Not yet implemented")
    }

    override fun setter(target: OfflinePlayer) {
        TODO("Not yet implemented")
    }
}